http://lists.freedesktop.org/archives/compiz/2007-October/002697.html

From: Danny Baumann <dannybaumann@web.de>
Date: Tue, 2 Oct 2007 05:16:00 +0000 (+0200)
Subject: We always need to restack windows on map. Newly created windows are placed on top ...
X-Git-Tag: start
X-Git-Url: http://gitweb.freedesktop.org/?p=xorg/app/compiz.git;a=commitdiff;h=b6c6acc70261d0942977441914c23e7a1c99215e

We always need to restack windows on map. Newly created windows are placed on top of the stack first, which is not desired for some windows that may be denied focus (such as desktop windows).
This commit reverts commit 67648904b15c7781000a212a122b558a2d2afcf6.
---

--- a/src/event.c
+++ b/src/event.c
@@ -1922,10 +1922,7 @@ handleEvent (CompDisplay *d,
 
 		allowFocus = allowWindowFocus (w);
 
-		if (allowFocus)
-		    updateWindowAttributes (w, CompStackingUpdateModeNormal);
-		else
-		    updateWindowAttributes (w, CompStackingUpdateModeNone);
+		updateWindowAttributes (w, CompStackingUpdateModeNormal);
 
 		if (!allowFocus &&
 		    (w->type & ~(CompWindowTypeSplashMask |
